app/models/review.rb

```diff
@@ -0,0 +1,15 @@
+class Review < ApplicationRecord
+  belongs_to :art
+  belongs_to :user
+
+  validates :rating, presence: true, inclusion: 1..5, numericality: { only_integer: true }
+  validate :user_has_booked_before
+
+  private
+
+  def user_has_booked_before
+    return if user.bookings.include?(art)
+
+    errors.add(:user, "cannot review a listing he has never booked")
+  end
```

Comment:
user.bookings is still just an array of Booking instances but you want to compare an Art instances so that's not gonna work.
You need to do `user.bookings.map(&:art).include?(art)` instead

`.map(&art)` is the same as `.map { |booking| booking.art }` which means that from an array of Booking instances you then get an array of Art instances